mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1640,6 +1720,56 @@ pub fn check_all_inputs_signed_by_pub(tx: &UtxoTx, expected_pub: &[u8]) -> Resul
     Ok(true)
 }
 
+pub fn watcher_validate_taker_fee<T: UtxoCommonOps>(
+    coin: T,
+    taker_fee_hash: Vec<u8>,
+    verified_pub: Vec<u8>,
+) -> ValidatePaymentFut<()> {
+    let fut = async move {
+        let mut attempts = 0;
+        let taker_fee_hash = H256Json::from(taker_fee_hash.as_slice());
+        loop {
+            let taker_fee_tx = match coin
+                .as_ref()
+                .rpc_client
+                .get_transaction_bytes(&taker_fee_hash)
+                .compat()
+                .await
+            {
+                Ok(t) => t,
+                Err(e) => {
+                    if attempts > 2 {
```

Comment:
I think the attempt should be >> than 2 for better assurance...what do you think??

Reply:
I think getting an error 3 times consecutively is enough to show that there's an issue which will likely to repeat in the next attempts, and trying it further would just cost rpc requests without success.

Reply:
ok note. thanks